src/vm/ceeload.cpp

```diff
@@ -11817,6 +11820,63 @@ static bool GetBasename(LPCWSTR _src, __out_ecount(dstlen) __out_z LPWSTR _dst,
     return true;
 }
 
+static LPCWSTR s_pCommandLine = NULL;
+
+// Rerieve the full command line for the current process.
+LPCWSTR GetManagedCommandLine()
+{
+    LIMITED_METHOD_CONTRACT;
+    return s_pCommandLine;
+}
+
+void SaveManagedCommandLine(LPCWSTR pwzAssemblyPath, int argc, LPCWSTR *argv)
+{
+    CONTRACTL
+    {
+        NOTHROW;
+        GC_NOTRIGGER;
+        MODE_ANY;
+    }
+    CONTRACTL_END;
+
+    // Get the command line.
+    LPCWSTR osCommandLine = GetCommandLineW();
+
+#ifndef FEATURE_PAL
+    // On Windows, osCommandLine contains the executable and all arguments.
+    s_pCommandLine = osCommandLine;
+#else
+    // On UNIX, the PAL doesn't have the command line arguments, so we must build the command line.
+    // osCommandLine contains the full path to the executable.
+
+    SIZE_T commandLineLen = 0;
+    
+    for (int i = 0; i < argc; i++)
+    {
+        commandLineLen += (wcslen(argv[i]) + 1);
+    }
+
+    // Allocate a new string for the command line.
+    WCHAR *pCommandLine = new WCHAR[commandLineLen + 1];
+    SIZE_T remainingLen = commandLineLen ;
+    WCHAR *pCursor = pCommandLine;
+
+    for (int i = 0; i < argc; i++)
+    {
+        SIZE_T curLen = wcslen(argv[i]);
+        wcscpy_s(pCursor, remainingLen, argv[i]);
+        pCursor += curLen;
+        remainingLen -= curLen;
+
+        wcscpy_s(pCursor, remainingLen, W(" "));
+        pCursor += 1;
+        remainingLen -= 1;
+    }
+
+    s_pCommandLine = pCommandLine;
```

Comment:
Shouldn't `s_pCommandLine` be deallocated somewhere in shutdown, like we had on `EventPipe`?
```cpp
#ifdef FEATURE_PAL
    delete[] s_pCommandLine; 
    s_pCommandLine = NULL;
#endif
```

Reply:
Perhaps,  I will consider where this can be put.